api/src/main/java/org/eclipse/microprofile/rest/client/ext/ResponseExceptionMapper.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2017 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.rest.client.ext;
+
+import javax.annotation.Priority;
+import javax.ws.rs.core.Response;
+import java.util.Optional;
+
+/**
+ * Converts an JAX-RS Response object into an Exception.
+ *
+ */
+public interface ResponseExceptionMapper {
+    int DEFAULT_PRIORITY = 100;
+
+    /**
+     * Converts a given Response into a Throwable.  The runtime will throw this if it is present.
+     * @param response the JAX-RS response processed from the underlying client
```

Comment:
Please add the following from the spec to the javadoc too:
> If this method reads the response body as a stream it must ensure that it resets the stream.

Reply:
Technically it should be both of the methods.  But yes, will add.

Reply:
@OndrejM its been added if you can look.